Sources/Nimble/Matchers/PostNotification.swift

```diff
@@ -15,10 +15,25 @@ internal class NotificationCollector {
     }
 
     func startObserving() {
-        self.token = self.notificationCenter.addObserver(forName: nil, object: nil, queue: nil) { [weak self] n in
-            // linux-swift gets confused by .append(n)
-            self?.observedNotifications.append(n)
-        }
+        #if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+            self.token = self.notificationCenter.addObserver(forName: nil, object: nil, queue: nil) { [weak self] n in
+                // linux-swift gets confused by .append(n)
+                self?.observedNotifications.append(n)
+            }
+        #elseif swift(>=4.0.2) || swift(>=3.2.2)
```

Comment:
Unfortunately we can't use `#if if swift(>=x.y.z)` 😞 

https://travis-ci.org/Quick/Nimble/jobs/303516955